### DISCUSSION TITLE
How to handle debug fission

### DISCUSSION THREAD

**Discussion by jeremy-rifkin** [davea42/libdwarf-code#244] [Q&A] [answered]
I'm trying to add support for debug fission/split dwarf to a library I work on for stack tracing. I'm encountering a lot of challenges along the way. For example, if in a split dwarf .dwo file `DW_AT_low_pc` is an indexed address into the `.debug_addr` table from the linked binary. When I encounter this in libdwarf I get `dwarf error 257 DW_DLE_MISSING_NEEDED_DEBUG_ADDR_SECTION(257)`. I can check the form of the attribute before calling `dwarf_lowpc` and do this resolution myself, however, does libdwarf have a way to provide a reference to the main executable's address table? Does libdwarf have any utilities that can make working with split dwarf easier?

**Comment by jeremy-rifkin** [comment]
Ah, I found `dwarf_set_tied_dbg` in the documentation which appears to be exactly what I am looking for.

**Comment by davea42** [comment, marked as answer]
Jeremy answered this himself.  dwarf_set_tied_dbg().
